File: Source/Camping Mod 2.0/common/rikmuld/camping/inventory/slot/SlotCooking.java
package rikmuld.camping.inventory.slot;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;
import rikmuld.camping.entity.tileentity.TileEntityCampfireCook;
import rikmuld.camping.misc.cooking.CookingEquipment;

public class SlotCooking extends Slot {

	public boolean active;
	public CookingEquipment equipment;
	public TileEntityCampfireCook fire;

	public SlotCooking(IInventory par1iInventory, int id, int x, int y)
	{
		super(par1iInventory, id, x, y);
		deActivate();
	}

	public void activate(int x, int y, CookingEquipment equipment, TileEntityCampfireCook fire)
	{
		active = true;
		this.equipment = equipment;
		this.fire = fire;

		xDisplayPosition = x;
		yDisplayPosition = y;
	}

	public void deActivate()
	{
		active = false;
		equipment = null;
		fire = null;

		xDisplayPosition = -1000;
		yDisplayPosition = -1000;
	}

	@Override
	public int getSlotStackLimit()
	{
		return 1;
	}

	@Override
	public boolean isItemValid(ItemStack stack)
	{
		return (equipment != null) && (fire != null)? equipment.canCook(stack.itemID, stack.getItemDamage()) && (fire.getStackInSlot(12) != null)? true:(equipment.getCookedFood(stack.itemID, stack.getItemDamage()) != null):false;
	}
}

File: scala/com/rikmuld/camping/features/blocks/campfire/cook/SlotCooking.scala
package com.rikmuld.camping.features.blocks.campfire.cook

import com.rikmuld.camping.features.blocks.campfire.cook.equipment.CookingEquipment
import net.minecraft.inventory.{IInventory, Slot}
import net.minecraft.item.ItemStack

class SlotCooking(inv: IInventory, id: Int, x: Int, y: Int) extends Slot(inv, id, x, y) {
  var active: Boolean =
    false

  var equipment: Option[CookingEquipment] =
    None

  var fire: Option[TileEntityCampfireCook] =
    None

  deActivate()

  def activate(x: Int, y: Int, equip: CookingEquipment, tile: TileEntityCampfireCook): Unit = {
    active = true

    equipment = Some(equip)
    fire = Some(tile)

    xPos = x
    yPos = y
  }

  def deActivate(): Unit = {
    active = false

    equipment = None
    fire = None

    xPos = -1000
    yPos = -1000
  }

  override def getSlotStackLimit: Int =
    1

  override def isItemValid(stack: ItemStack): Boolean =
    equipment.exists(_.canCook(stack))
}